src/stores/ontRuns.js

```diff
@@ -0,0 +1,171 @@
+import { defineStore } from 'pinia'
+import { handleResponse } from '@/api/ResponseHelper'
+import useRootStore from '@/stores'
+import useOntRootStore from '@/stores/ontRoot'
+import store from '@/store'
+
+export const useOntRunsStore = defineStore('ontRuns', {
+  state: () => ({
+    currentRun: {
+      flowcell_attributes: [],
+      id: 'new',
+      state: null,
+      instrument_name: null,
+    },
+  }),
+  getters: {
+    runRequest: () => {
+      const rootStore = useRootStore()
+      return rootStore.api.traction.ont.runs
+    },
+  },
+  actions: {
+    newRun() {
+      this.currentRun = {
+        id: 'new',
+        instrument_name: null,
+        state: null,
+        flowcell_attributes: [],
+      }
+    },
+    async createRun() {
+      const ontRootStore = useOntRootStore()
+      const run = this.currentRun
+      const request = this.runRequest
+
+      const existingInstruments = ontRootStore.instruments
+      const instrument_id = existingInstruments.find((i) => i.name == run.instrument_name).id
+
+      //TODO: This need to be refactored to use the Pinia once ont/ppols is migrated
+      const existingPools = store.getters['traction/ont/pools/pools']
+
+      const flowcell_attributes = run.flowcell_attributes
```

Comment:
it looks like createRun and updateRun are also using complex methods to create stuff and doing the same thing which could be resolved by using types.m I am also not sure what fc is.